### ISSUE TITLE
Seg Fault on SetPredictionType

### ISSUE THREAD

**Issue by MartinBrugnara** [catboost/catboost#1748]
Problem: Invocation of SetPredictionType() with any EApiPredictionType value from C (`c_api.h`) causes Segmentation Fault.
catboost version: f1ed552f10bac34b6bff8f0ac97b4c89189b57d4
Operating System: Ubuntu 18, Linux 5.4.0-74
CPU: Intel 

cc/ @CristianCantoro

**Comment by markRtron** [comment]
Hey there,

had the same issue and debugged it with GDB. The problem is, that the implementation of SetPredictionType is faulty. It does not check, whether Evaluator is set (in contrast to the Calc functions) or not. There are two ways to fix this problem:

1. Call any function which makes use GetCurrentEvaluator like Calc before you call SetPredictionType. This will init the Evaluator correctly and a subsequent call to SetPredictionType will work. Although this is not really nice, you do not have to recompile the catboost.
2. Fix the problem in catboost/catboost/libs/model/model.h:675 by applying the following patch:
```
diff --git a/catboost/libs/model/model.h b/catboost/libs/model/model.h
index c6a6a86290..2bcb060274 100644
--- a/catboost/libs/model/model.h
+++ b/catboost/libs/model/model.h
@@ -672,6 +672,9 @@ public:
 
     void SetPredictionType(NCB::NModelEvaluation::EPredictionType predictionType) const {
         with_lock(CurrentEvaluatorLock) {
+            if (!Evaluator) {
+                Evaluator = NCB::NModelEvaluation::CreateEvaluator(FormulaEvaluatorType, *this);
+            }
             Evaluator->SetPredictionType(predictionType);
         }
     }
```
This adds the missing check right before using the pointer. Hope this helps..

**Comment by LyzhinIvan** [comment]
Thanks, @markRtron! I have committed your fix in 49570244b1638849b876ace55cc0285afdc6f621. Now it works.
